### PR TITLE
Create Agents, Personas and Contributors files

### LINKED ISSUE #16319: Create repo AI helper file
Files
- `agents.md` Core ai helper file 
  - https://agents.md /https://github.com/agentsmd/agents.md
  - we could ignore content suggestions and stick with the naming convention
- `.github/copilot-instructions.md` --> symlink to agents.md
- `GEMINI.md` --> symlink to agents.md
- `CLAUDE.md` --> symlink to agents.md

TBD - split agent into individual files for different personas, or include different personas in the same root file?

Contents
- Can this itself be created via AI? 
- To build up over time, but generally
  - Tech Stack: Which versions of frameworks are being used.
  - Coding Standards: Preference for functional vs. object-oriented patterns.
  - Build/Test Commands: Exactly how to run the code so the agent can verify its own changes.
  - Architecture: A high-level overview of how data flows through the system.
- Specifically
  - Lint info
  - use scss
  - how to write unit tests
  - vue components have script before template
  - when creating pr's they should link to the issue they resolve (for co-pilot...)

Guides/Notes
- https://github.blog/ai-and-ml/github-copilot/how-to-write-a-great-agents-md-lessons-from-over-2500-repositories/ (Note - this makes use of sub-agents)
- https://milvus.io/ai-quick-reference/where-are-the-gemini-cli-config-files-stored

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
Fixes #16319

### Occurred changes and/or fixed issues
- Create a root AGENTS.md to be used by all AI tools
  - There's a lot of TBD, more info in docs/agents.md/README 
- Populate with three categories
  - agents
  - contributors
  - personas
- Create symlinks to fix (claude) or assist (gemini) find the info
  - Gemini VS Code - Didn't need any additional pointers, used AGENTS and raw files
  - Gemini CLI - Needed gemini.md for some types of requests
  - Copilot Github activities - nothing added
  - Copilot CLI - nothing added
  - Claude - Added CLAUDE.md

### Areas or cases that should be tested

We should try to check that each tool correctly finds + uses info from agents

- Gemini VS Code - Didn't need any additional pointers, used AGENTS and raw files
- Gemini CLI - Needed gemini.md for some types of requests
- Copilot Github activities - TBD
- Copilot CLI - TBD
- Claude - TBD

I used the following prompts to test. They only ask for information that is within the files instead of executing actions that use the information in the result

- `what's this project about?`
- `can you tell me the process of fixing a github issue for this repo?`
- `what makes a good software developer?`
- `what should never be done in this project?`
- `what are the folders in this project for?`
- `what are the env vars i could use when running e2e tests?`

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [x] The PR has been reviewed in terms of Accessibility
- [x] The PR has considered, and if applicable tested with, the three Global Roles `Admin`, `Standard User` and `User Base`

### CODE REVIEW

**Comment by richard-cox** [comment]
@codyrancher , @rak-phillip , @marcelofukumoto This is the start of our agent, etc files. We'll add and expand these but hopefully we can get this merged soon. ATM looking for initial opinions before it's ready for a final review

TODO
- ~Confirm that the agents file is successfully being considered by AIs~
  - i'll need assistance with claude and copilot
- ~Confirm that the process works for everyone~
- ~Add a guide somewhere that explains files + process~

File: agents.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+## Tools
+
+- **Install dependencies**: `yarn install`
+- **Start development server**: `API=<Rancher_Backend_URL> yarn dev`
+  - The `API` environment variable should point to a running Rancher server (e.g., `https://localhost`).
+  - The dashboard will be available at `https://127.0.0.1:8005`.
+- **Build**: `yarn build`
+- **Lint**: `yarn lint`
+- **Unit Tests**: `yarn test` (Jest)
+- **E2E Tests**: Requires configuration
```
Comment (codyrancher) [inline agents.md]: I've had some luck with this format specifically for tests. I didn't use the same tool format.
> # Testing
> ## Unit Tests
> - To run a single test file: `yarn test <full-path-to-test-file>`
> 
> ## E2E Tests (Cypress)
> - Interactive mode: `yarn cy:e2e`
> - Headless mode: `yarn cy:run`
> - Run a specific spec file: `yarn cy:run --spec cypress/e2e/tests/<path-to-spec>.spec.ts`
> 
> ### Required Environment Variables
> - `TEST_PASSWORD` - Password for login (or `CATTLE_BOOTSTRAP_PASSWORD` for setup tests)
> - `TEST_BASE_URL` - Dashboard URL (defaults to `https://localhost:8005`)
> - `TEST_USERNAME` - Username (defaults to `admin`)
> 
> ### Optional Environment Variables
> - `TEST_SKIP` - Comma-separated test directories to skip (e.g., `setup,extensions`)
> - `TEST_ONLY` - Comma-separated test directories to run exclusively
> - `GREP_TAGS` - Filter tests by tags

And then I have my configuration defined in .env. 

I will note that I've watched the agent use creds in the .env files in unspecified ways so either finding a different way or being careful is still needed.

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- **Lint**: `yarn lint`
+- **Unit Tests**: `yarn test` (Jest)
+- **E2E Tests**: Requires configuration
+
+## Boundaries
+
+- ✅ **Always:** 
+  - Write to `creators/`, `cypress/`, `docs/`, `docusaurus/`, `pkg/`, `scripts/`, `shell/` and `storybook/`.
+  - Run tests before commits.
+  - Follow existing naming conventions (PascalCase for components, camelCase for functions).
```
Comment (codyrancher) [inline AGENTS.md:30]: I've broken this out into a separate section for myself and while I can get it to style things correctly I've mostly fallen back on.

> # Code Style
>- After editing or saving a file make sure it's automatically formatted with eslint

File: agents.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# AGENTS.md
+
+> `/agents.md` should be generated by running `scripts/generate-agent-and-persona-mds.sh` script which inserts personas and contributors
+
+## Project Overview
+
+Rancher Dashboard is the user interface for Rancher, built using Vue.js and TypeScript. It interacts with the Rancher API to manage Kubernetes clusters.
+
+## Personas
```
Comment (codyrancher) [inline agents.md]: I haven't really tried out personas. I'll have to try it out. 

The closest analogue for me is listing preferences and architectural design choices. i.e.

> - We should favor many small components instead of large pages with copious styling

Or guides to get things started:

> # Creating Git Branches
> - Create a branch for an issue: `git checkout master && git checkout -b issue-${issueNumber}`

File: agents.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+## Boundaries
+
+- ✅ **Always:** 
+  - Write to `creators/`, `cypress/`, `docs/`, `docusaurus/`, `pkg/`, `scripts/`, `shell/` and `storybook/`.
+  - Run tests before commits.
+  - Follow existing naming conventions (PascalCase for components, camelCase for functions).
+- ⚠️ **Ask first:** Adding dependencies
+- 🚫 **Never:** 
+  - Commit secrets, `.env`, or API keys.
```
Comment (codyrancher) [inline agents.md]: Probably want to mention logging or something as well. I've seen debug logs with secrets but thus far only in backend code.

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+- ⚠️ **Ask first:** Adding dependencies
+- 🚫 **Never:** 
+  - Commit secrets, `.env`, or API keys.
+  - Edit `node_modules/`.
+  - Commit directly to `master` (use PRs).
```
Comment (codyrancher) [inline AGENTS.md:37]: is this because of copilot? I'd think we'd want separate branches when developing locally.

File: AGENTS.md (hunk 2 above)
Comment (codyrancher) [reply AGENTS.md:30]: I did just notice all the other styling directives below.

File: AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+- If backports are needed they can be made via the backport bot
+  - github issue
+    - `/backport <target milestone>` e.g `/backport v2.12.2`
+  - pull requests
+    - `/backport <target milestone> <target branch>` e.g. `/backport v2.12.2 release-12`
+    - All backported pull requests must link to a backported issue
+
+### Creating a commit
+
+- Follow the [Chris Beams](http://chris.beams.io/posts/git-commit/) 'seven rules of a great Git commit message'  for commit messages.
```
Comment (codyrancher) [inline AGENTS.md:136]: I like this

File: AGENTS.md (hunk 2 above)
Comment (richard-cox) [reply AGENTS.md:30]: I was split where to add this one. In the end I've put it within the agents boundaries section given it's a bit more direct to AI. 

We can also update the contributors guide with similar but more information for things like configuring vscode to apply on save?

File: agents.md (hunk 3 above)
Comment (richard-cox) [reply agents.md]: > We should favor many small components instead of large pages with copious styling

I've added something that should cover this into the contributors code styles and standards part

`Large pages with lots of code and styles should be avoided by breaking the page up into smaller Vue components.`

> Create a branch for an issue: git checkout master && git checkout -b issue-${issueNumber}

I got stuck with this one, mainly trying to succinctly describe the target branch (master, release-X).

See contributors creating a branch --> to resolve an issue for attempt

File: AGENTS.md (hunk 5 above)
Comment (richard-cox) [reply AGENTS.md:37]: i got this from somewhere, but yep, it's to prevent agents committing to master. I'll put the required behaviour in  the always section

File: agents.md (hunk 1 above)
Comment (richard-cox) [reply agents.md]: I've added this to the contributors guide with a link to our actual docs

The idea is that its there for AI when needed, but humans can get more info over in storybook doc world

File: agents.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+### Software Developer
+
+You are an expert Senior Software Engineer specializing in Vue.js and TypeScript. You have deep knowledge of Kubernetes and the Rancher ecosystem.
+- **Focus**: Writing clean, maintainable, and performant code.
+- **Priorities**: Adhering to the project's code style, ensuring type safety, and following best practices for component design.
+
+### QA Software Developer
+
+You are a meticulous QA Automation Engineer with expertise in ???, Cypress and end-to-end testing for complex web applications.
```
Comment (marcelofukumoto) [inline agents.md]: Is this ??? correct? I imagined it would use as a dynamic input so I wonder what would be the expectation here.

File: agents.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
+# AGENTS.md
+
+> `/agents.md` should be generated by running `scripts/generate-agent-and-persona-mds.sh` script which inserts personas and contributors
```
Comment (marcelofukumoto) [inline agents.md]: So we need to keep only one file?
Using github copilot instructions I believe we can have multiple files and different contributors.
Is there an easier way to do that than keeping only one agents.md file?

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+- 🚫 **Never:** 
+  - Commit or log secrets, `.env`, or API keys.
+  - Edit `node_modules/`.
+  - Commit directly to `master` (use PRs).
+
+## Contributors Guide
+
+### Getting Started
+
+Please see the [Rancher UI Internal Documentation](https://extensions.rancher.io/internal/docs).
```
Comment (marcelofukumoto) [inline AGENTS.md:55]: I wonder how reliable the AI can use "links". I remember having problem with that. But would be nice to check! Specially because if that is the case, would feel interesting to focus on some nice "guides" outside the repository and use as best practices. Liked this one.

File: AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+    - SCSS variables are in `shell/assets/styles/`.
+- **File Structure:**
+  - `creators/`: Tools for scaffolding new extensions.
+  - `cypress/`: Cypress E2E test specifications.
+  - `docs/`: Internal documentation source.
+  - `docusaurus/`: Public documentation source.
+  - `pkg/`: Internal Rancher UI extensions.
+  - `scripts/`: Bash scripts used in build, test and github workflows
+  - `shell/`: Core application logic, components, and pages.
+  - `storybook/`: Component documentation source
```
Comment (marcelofukumoto) [inline AGENTS.md:84]: Would be nice to have some. "good examples"
From my previous experience that was the best way of using AI reliably. Give some examples of good code.
I like the list here!

File: AGENTS.md (hunk 6 above)
Comment (marcelofukumoto) [reply AGENTS.md:136]: Really nice! I wonder if it can reliably use references from the page.

File: contributors.md [hunk 11]
```diff
@@ -0,0 +1 @@
+Contributors guide can be found in [docs/contributors/contributors.md](docs/contributors/contributors.md)
```
Comment (marcelofukumoto) [inline contributors.md]: I tried to use the AI using the configs. I didn't see the contributors being used as reference. I wonder how that one works. Does work well?

File: scripts/generate-agent-and-persona-mds.sh [hunk 12]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+# Get the directory of the script
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# Project root is one level up from scripts/
+PROJECT_ROOT="$(dirname "$SCRIPT_DIR")"
+
+# Source files
+ROOT_AGENTS="${PROJECT_ROOT}/docs/agents/root_agents.md"
```
Comment (marcelofukumoto) [inline scripts/generate-agent-and-persona-mds.sh]: Would not be better for the create to use every file from the agents folder instead.

File: agents.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+
+## Contributors Guide
+
+### Getting Started
+
+Please see the [Rancher UI Internal Documentation](https://extensions.rancher.io/internal/docs).
+
+To get started, follow the `Getting Started` section.
+
+### Project Information
```
Comment (marcelofukumoto) [inline agents.md]: I feel those Project Information belongs to the README file instead of agents.md.
On my perspective the agents.md should have mostly the "HOW TO" resolve some of the AI inputs, project information helps, but it helps on the README file as well.

File: agents.md (hunk 7 above)
Comment (richard-cox) [reply agents.md]: Ah i should have put a TODO: RC in there, will update

File: agents.md (hunk 8 above)
Comment (richard-cox) [reply agents.md]: > So we need to keep only one file?

No, we have multiple files that are then brought into one file.

> Using github copilot instructions I believe we can have multiple files and different contributors.

That is the case, as per gemini and a few others, however they don't all use the same files names and locations.

The only commonality i've found is that root agent file. I think most will also look for an agent in sub folders as well, but not sure how / when those are picked up by AI. I plan to do some testing on this to ensure that whats currently in the agents.md is applied... and also when/how subfolders are considered.

File: AGENTS.md (hunk 9 above)
Comment (richard-cox) [reply AGENTS.md:55]: I don't think they do at all, which is why the script brings in files rather than references them.

This one in particular is for humans brought in via the contributors guide. There's a bit more info in [this](https://github.com/rancher/dashboard/pull/16388#discussion_r2717324662) comment

File: AGENTS.md (hunk 10 above)
Comment (richard-cox) [reply AGENTS.md:84]: I think we'll need a separate guide for the storybook stuff, will create an issue for it

File: contributors.md (hunk 11 above)
Comment (richard-cox) [reply contributors.md]: This is for humans.

The docs contributors guide is copied into the agents file by the script, so it would be worrying if the agents file was ignored.

If it turns out all agents consider the root contributors file we can just move the docs one to the root

File: scripts/generate-agent-and-persona-mds.sh (hunk 12 above)
Comment (richard-cox) [reply scripts/generate-agent-and-persona-mds.sh]: Possibly. At the moment they're inserted into specific places in the root agents file.

I think we could either work with  
- explicit files
- files in this folder
- find all agent.md files and move into the root md file 

For the last two we should work out how to format them such that the flow isn't confusing and reduces overlap in sections

File: agents.md (hunk 13 above)
Comment (richard-cox) [reply agents.md]: This kind of info (tech stacks, specific tech advice, location descriptions) was recommended by a few places for good agent file content.

This comes in via the contributors guide.

File: AGENTS.md (hunk 10 above)
Comment (marcelofukumoto) [reply AGENTS.md:84]: Oh, not only for storybook things. And maybe not for now as well. 
So just a comment for us to remember that examples are a really good way of adding depth to each one of paths or files. 
I saw that we can have AGENTS.MD on each folder. so I guess we will be having those on each one.
So will be good to have a ticket for the storybook already! Thanks!

File: AGENTS.md (hunk 9 above)
Comment (marcelofukumoto) [reply AGENTS.md:55]: Oh ok. I saw some instructions and other configs online having LINKs as well. I never tried to use it because I remember testing and didnt work well, so decided to get all files in instead.
I was wondering if working well now, but maybe not. Not sure to be honest.
But as you said, it is for humans, so great!

File: agents.md (hunk 13 above)
Comment (marcelofukumoto) [reply agents.md]: That is interesting. I heard the opposite before.
Ok! All good !

File: scripts/generate-agent-and-persona-mds.sh (hunk 12 above)
Comment (marcelofukumoto) [reply scripts/generate-agent-and-persona-mds.sh]: Ok. I think explicit files would work well enough as well.
I guess the question would be more what was the plan for more files. 
So all good!

File: contributors.md (hunk 11 above)
Comment (marcelofukumoto) [reply contributors.md]: Oh! That is true, you are copying the content. I thought it was an alternative to agents.md.
From what I tested on gemini and copilot it uses the agents.md and the content of contributors is copied to agents. All good.

File: agents.md (hunk 8 above)
Comment (marcelofukumoto) [reply agents.md]: I am worried of the agents.md file growing a lot, but from what you shared the actual files to edit would be separated anyway. So the agents.md will be just a compilation of all the other files.
I wonder if having everything inside one file hinders the way the AI uses the instructions inside it, but considering the popularity of AGENTS.MD, if does have a problem it will bring a solution at some point.

One thing that I couldn't see the AGENTS.MD replicating is the use of the "agents".
With agents folder you can select the agent you want on the copilot for example. I wonder if AGENTS.MD replicates that. If doesn't if having a really big AGENTS.MD file causes any trouble. Or if there is a way to pick the correct instructions.

File: agents.md (hunk 7 above)
Comment (marcelofukumoto) [reply agents.md]: Ok!

**Comment by marcelofukumoto** [comment]
That is a great start.
We already have a way to have multiple files which will be explicit added to the script.
We have some great initial explanation about the project and some specific rules and personas.
It seems the AGENTS.MD is how we want to move forward with it.

**Comment by richard-cox** [comment]
@marcelofukumoto + @codyrancher I've updated the process following some discussions and review comments. Big highlights

- created folders for agents, contributors and personas. each folder contains files. each file will be inserted into the agents.md. this allows for much more flexibility
- update the agents README with more info, including future tasks (which includes info from PR comments)
- rename agents.md to AGENTS.md
- removed ./contributors and ./personas (both still included in agents)
- added a yarn target for the script
- added symlink file claude.md pointing to agents file (which i think is the only big one that doesn't use it)

I'll next concentrate if tools actually pick this up, then hopefully we can get this in

**Comment by richard-cox** [comment]
I've validated that gemini at least uses the agents file for information, if not when solving issues (the cli side was a bit iffy, so include a symlnk for gemini specific config file --> agents file).

I haven't done anything with co-pilot though, does any reviewer here have that setup?

File: docs/agents.md/contributors/resolving_gh_issues.md [hunk 14]
```diff
@@ -0,0 +1,36 @@
+## Milestone guidance
+
+- All issues must first be resolved in the `master` branch
+- If backports are needed they can be made via the backport bot
+  - github issue
+    - `/backport <target milestone>` e.g `/backport v2.12.2`
+  - pull requests
+    - `/backport <target milestone> <target branch>` e.g. `/backport v2.12.2 release-12`
```
Comment (rak-phillip) [inline docs/agents.md/contributors/resolving_gh_issues.md]:
```suggestion
    - `/backport <target milestone> <target branch>` e.g. `/backport v2.12.2 release-2.12`
```

**Comment by marcelofukumoto** [comment]
I tested using COPILOT. Let me add my test here:
LLM: Claude Sonnet 4
Prompt: Who are the personas setup for the AI?
Response:
```
Looking at the [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) file that's currently open, I can see the personas that are set up for the AI. There are two main personas defined:

1. QA Software Developer
Role: Meticulous QA Automation Engineer with expertise in Cypress and end-to-end testing for complex web applications
Focus: Ensuring reliability, stability, and comprehensive test coverage
Priorities: Writing robust Cypress tests, identifying edge cases, and verifying that UI changes do not introduce regressions
2. Software Developer
Role: Expert Senior Software Engineer specializing in Vue.js and TypeScript with deep knowledge of Kubernetes and the Rancher ecosystem
Focus: Writing clean, maintainable, and performant code
Priorities: Adhering to the project's code style, ensuring type safety, and following best practices for component design
These personas are designed to help AI agents understand their roles when working on the Rancher Dashboard project, which is a Vue.js/TypeScript application for managing Kubernetes clusters. The QA persona focuses on testing and quality assurance, while the Software Developer persona focuses on development and code quality.
```

It used the AGENTS.md as reference.
image

----

Prompt: If I run an unit test which command I should execute?
```
Based on the [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) documentation, to run unit tests you should execute:

This runs the Jest unit tests for the project.

If you want to run a single test file, you can use:

For example:

The project uses Jest as the testing framework for unit tests.
```

---

Prompt: Fix this file based on the best practices, explain.
Added file as context: shell/edit/pod.vue

```
Looking at this Vue component, I can see several areas that need improvement based on Vue.js and project best practices. Let me fix them:

pod.vue+13-21
Let me fix the linting errors:

pod.vue+1-3
Explanation of Changes:
Here are the improvements I made to align with Vue.js and project best practices:

1. Composition API over Options API
Converted from Options API to Composition API using <script setup> which is the preferred approach according to the project guidelines
This provides better TypeScript support and is more concise
2. TypeScript Integration
Added lang="ts" to the script tag for proper TypeScript support
Defined proper TypeScript interfaces for props with Props interface
Used defineProps<Props>() for type-safe props definition
3. Removed Unnecessary Mixins
Removed CreateEditView, FormValidation, and WorkLoadMixin mixins since this component is just a simple wrapper that passes all props to the Workload component
The actual logic should be handled in the Workload component itself
4. Improved Props Handling
Used withDefaults() for setting default values in a type-safe way
Changed v-bind="$props" to v-bind="props" to use the Composition API props
5. Component Structure
Followed the recommended order: <script> → <template> → <style>
Added the required <style lang="scss" scoped> section as per project guidelines
6. Removed Redundant Component Name
The component name was misleading (WorkloadDeployments for a pod component) and isn't necessary in <script setup>
This refactor makes the component cleaner, more type-safe, and follows modern Vue.js patterns while adhering to the project's coding standards.
```

---

It fixed the code, than:

---
Prompt: Are those changes based on the AGENTS.MD file?

```
Yes, the changes I made are directly based on the guidelines specified in the [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) file. Let me point out the specific connections:

From [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) - Code Style and Standards:
1. "TypeScript is preferred for new code"
✅ Added lang="ts" to the script tag
✅ Used TypeScript interfaces and proper type definitions
2. "Composition API components are preferred over Options API"
✅ Converted from Options API (export default { ... }) to Composition API (<script setup>)
3. "Place source tag above template above style"
✅ Followed the exact order: <script> → <template> → <style>
4. "style tag should contain lang='scss' scoped"
✅ Added <style lang="scss" scoped> section
From [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) - Boundaries - Always:
"Follow existing naming conventions"
✅ Used proper TypeScript conventions with PascalCase for the interface (Props)
✅ Used camelCase for the props constant
"After changing a vue, js or ts file make sure it's automatically formatted with eslint"
✅ The changes follow ESLint formatting rules (I fixed the linting errors that appeared)
The refactoring was specifically designed to align with the project's stated preferences for modern Vue.js development practices as outlined in the [AGENTS.md](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) documentation. Every change directly corresponds to a best practice mentioned in that file.
```

**Comment by richard-cox** [comment]
@codyrancher @marcelofukumoto @rak-phillip Think this is ready to go in now.

Gemini + Copilot should be picking up the changes. There's some pending tasks in docs/agents.md/README that we can run through after merging

**Review by marcelofukumoto — APPROVED** [review]
LGTM
The new file separation is great
The text has information related directly to the project
There is a script to add the text to the agents.md

Tested on copilot, worked well.

File: docs/agents.md/agents/boundaries.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
+## Boundaries
+
+- ✅ **Always:**
+  - Write to `creators/`, `cypress/`, `docs/`, `docusaurus/`, `pkg/`, `scripts/`, `shell/` and `storybook/`.
+  - Make commits in a new branch (for a PR).
```
Comment (momesgin) [inline docs/agents.md/agents/boundaries.md:5]: I'm not sure about this, I'm not a fan of the agent automatically committing the changes it makes. I personally always prefer to commit the changes myself.

Comment (richard-cox) [reply docs/agents.md/agents/boundaries.md:5]: That's fine and can continue.

This line though is more about if a commit is made make it in a new branch, rather than if any changes are made make a commit and make it in a new branch.
